src/main/scala/com/cloudera/sparkts/DateTimeIndexUtils.scala

```diff
@@ -0,0 +1,223 @@
+/**
+ * Copyright (c) 2015, Cloudera, Inc. All Rights Reserved.
+ *
+ * Cloudera, Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"). You may not use this file except in
+ * compliance with the License. You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied. See the License for
+ * the specific language governing permissions and limitations under the
+ * License.
+ */
+
+package com.cloudera.sparkts
+
+import java.time.{ZonedDateTime, ZoneId}
+
+import scala.collection.mutable.{ListBuffer, PriorityQueue}
+
+private[sparkts] object DateTimeIndexUtils {
+  implicit val dateTimeIndexOrdering = new Ordering[DateTimeIndex] {
+    override def compare(x: DateTimeIndex, y: DateTimeIndex): Int = {
+      val c = x.first.compareTo(y.first)
+      if (c == 0) x.size.compareTo(y.size) else c
+    }
+  }
+
+  /**
+   * Given an array of indices that are sorted and non-overlapping,
+   * consecutive two indices a and b are merged into a new index d
+   * according to the following rules:
+   *
+   *  - a and b are both irregular -> d is irregular
+   *  - a and b are both of size 1 -> d is irregular
+   *  - either a is irregular and b is of size 1
+   *    or a is of size 1 and b is irregular -> d is irregular
+   */
+  def simplify(indices: Array[DateTimeIndex]): Array[DateTimeIndex] = {
+    val simplified = new ListBuffer[DateTimeIndex]
```

Comment:
> ArrayBuffer is preferable to ListBuffer for performance ~ @sryza 

Only append operations are performed and ```ListBuffer``` is better than ```ArrayBuffer``` in append.
http://docs.scala-lang.org/overviews/collections/performance-characteristics.html

Am I missing something?